samples/apps/cap/py/demo/CAPAutGenGroupDemo.py

```diff
@@ -0,0 +1,69 @@
+import time
+from autogen import AssistantAgent, GroupChat, GroupChatManager, UserProxyAgent, config_list_from_json
+from autogencap.DebugLog import Info
+from autogencap.LocalActorNetwork import LocalActorNetwork
+from autogencap.ag_adapter.AG2CAP import AG2CAP
+from autogencap.ag_adapter.CAP2AG import CAP2AG
+
+
+def cap_ag_group_demo():
+    config_list = config_list_from_json(env_or_file="OAI_CONFIG_LIST")
+    gpt4_config = {
+        "cache_seed": 73,
+        "temperature": 0,
+        "config_list": config_list,
+        "timeout": 120,
+    }
+
+    user_proxy = UserProxyAgent(
+        name="User_proxy",
+        system_message="A human admin.",
+        is_termination_msg=lambda x: "TERMINATE" in x.get("content"),
+        code_execution_config={
+            "last_n_messages": 2,
+            "work_dir": "groupchat",
+            "use_docker": False,
+        },
+        human_input_mode="TERMINATE",
+    )
+    coder = AssistantAgent(name="Coder", llm_config=gpt4_config)
+    pm = AssistantAgent(
+        name="Product_manager",
+        system_message="Creative in software product ideas.",
+        llm_config=gpt4_config,
+    )
+
+    # Composable Agent Network adapter
+
+    network = LocalActorNetwork()
+    user_proxy_cap2ag = CAP2AG(ag_agent=user_proxy, the_other_name="chat_manager", init_chat=True, self_recursive=False)
+
+    coder_cap2ag = CAP2AG(ag_agent=coder, the_other_name="chat_manager", init_chat=False, self_recursive=False)
+
+    pm_cap2ag = CAP2AG(ag_agent=pm, the_other_name="chat_manager", init_chat=False, self_recursive=False)
+    network.register(user_proxy_cap2ag)
+    network.register(coder_cap2ag)
+    network.register(pm_cap2ag)
+
+    user_proxy_ag2cap = AG2CAP(network, agent_name=user_proxy.name, agent_description=user_proxy.description)
```

Comment:
Is there a way to make this simpler from novice-to-intermediate user's perspective? Perhaps we can have two methods `network.register` which register an AutoGen agent, and `network.get_proxy` which returns an AutoGen agent? That way we hide `CAP2AG` and `AG2CAP`, and users only need to know: oh, I need to register agent to network, and oh, I can get a gateway proxy agent from the network. 

Also I think for this demo, even everything is in python, it would be great to split this into separate stacks, one for each agent, to show case the distributed nature of the setup. Otherwise it is not clear what the added value is.

Reply:
I created a new demo that demonstrates under standalone folder to demonstrates Agents running in different processes.